src/stacked_borrows.rs

```diff
@@ -338,16 +338,20 @@ impl<'tcx> Stack {
         &mut self,
         access: AccessKind,
         tag: SbTag,
-        dbg_ptr: Pointer<AllocId>, // just for debug printing amd error messages
+        (alloc_id, range): (AllocId, AllocRange), // just for debug printing amd error messages
         global: &GlobalState,
     ) -> InterpResult<'tcx> {
         // Two main steps: Find granting item, remove incompatible items above.
 
         // Step 1: Find granting item.
         let granting_idx = self.find_granting(access, tag).ok_or_else(|| {
             err_sb_ub(format!(
-                "no item granting {} to tag {:?} at {:?} found in borrow stack.",
-                access, tag, dbg_ptr,
+                "no item granting {} to tag {:?} exists in the borrow stack for memory range {}[{:#x}..{:#x}].",
```

Comment:
Potential alternative that seems more consistent with your other messages?
"trying {access} with tag {:?} at {:?}[{:#x}..{:#x}], but ,something something>"

In fact, even here it makes sense to distinguish "tag is in stack but is read-only", and "tag is not in stack", doesn't it?